dali/image/jpeg.cc

```diff
@@ -25,29 +25,30 @@ JpegImage::JpegImage(const uint8_t *encoded_buffer,
                      DALIImageType image_type)
   : GenericImage(encoded_buffer, length, image_type) {
 }
-#if !defined(DALI_USE_JPEG_TURBO)
-bool get_jpeg_size(const uint8 *data, size_t data_size, int *height, int *width) {
+#ifndef DALI_USE_JPEG_TURBO
+bool get_jpeg_size(const uint8 *data, size_t data_size, int *height, int *width, int *nchannels) {
   // Check for valid JPEG image
   unsigned int i = 0;  // Keeps track of the position within the file
   if (data[i] == 0xFF && data[i + 1] == 0xD8) {
     i += 4;
     // Retrieve the block length of the first block since
     // the first block will not contain the size of file
-    uint16_t block_length = data[i] * 256 + data[i + 1];
+    uint16_t block_length = (data[i] << 8) + data[i + 1];
```

Comment:
I would prefer to use '*' instead of `>>` or `<<`, as of someone silently change unsigned to signed of the target variable `>>` will start to behave not as expected, while the compiler is smart enough to optimize `*256` to appropriate shift operation.
@mzient - what is your opinion?

Reply:
We're not really multiplying here - we're inserting bytes to a desired position, so bit shift is IMO the right choice (BTW - left shift does not depend on signedness). What we _could_ change is to replace `+` with `|`, to emphasize that we're dealing with bits.

Reply:
For the record I plan to use the utilities in https://github.com/NVIDIA/DALI/pull/1310/files as soon as that PR gets merged